cscs-checks/libraries/io/hdf5_compile_run.py

```diff
@@ -19,10 +19,16 @@ def __init__(self, lang, linkage):
         self.descr = lang_names[lang] + ' HDF5 ' + linkage.capitalize()
         self.sourcepath = f'h5ex_d_chunk.{lang}'
         self.valid_systems = ['daint:gpu', 'daint:mc', 'dom:gpu', 'dom:mc']
-        self.valid_prog_environs = ['PrgEnv-cray', 'PrgEnv-gnu',
-                                    'PrgEnv-intel', 'PrgEnv-pgi']
         if linkage == 'dynamic':
-            self.valid_systems += ['eiger:mc']
+            self.valid_systems += ['eiger:mc', 'pilatus:mc']
+
+        # no cray-hdf5 as of PE 21.02 with PrgEnv-intel on Eiger and Pilatus
```

Comment:
there is:
```
/opt/cray/pe/hdf5/1.12.0.3/intel/19.0/
/opt/cray/pe/hdf5-parallel/1.12.0.3/intel/19.0/
```
on eiger. Is there an issue using them ?

Reply:
I would like to use them, how do you load the corresponding module though?
```
$ ml spider cray-hdf5/1.12.0.3

--------------------------------------------------------------------------------------------------------------------------------
  cray-hdf5: cray-hdf5/1.12.0.3
--------------------------------------------------------------------------------------------------------------------------------

    You will need to load all module(s) on any one of the lines below before the "cray-hdf5/1.12.0.3" module is available to load.

      aocc/2.2.0.1
      cce/11.0.4
      gcc/10.2.0
      gcc/9.3.0
```

Reply:
Ok, I see that `cray-hdf5-parallel` can be loaded with `intel`, but the serial `cray-hdf5` not (see above):
```$ ml spider cray-hdf5-parallel/1.12.0.3

--------------------------------------------------------------------------------------------------------------------------------
  cray-hdf5-parallel: cray-hdf5-parallel/1.12.0.3
--------------------------------------------------------------------------------------------------------------------------------

    You will need to load all module(s) on any one of the lines below before the "cray-hdf5-parallel/1.12.0.3" module is available to load.

      aocc/2.2.0.1  cray-mpich/8.1.4
      cce/11.0.4  cray-mpich/8.1.4
      craype-network-none  cray-mpich/8.1.4
      craype-network-ofi  cray-mpich/8.1.4
      craype-network-ucx  cray-mpich/8.1.4
      gcc/10.2.0  cray-mpich/8.1.4
      gcc/9.3.0  cray-mpich/8.1.4
      intel/19.1.3.304  cray-mpich/8.1.4
```

Reply:
Style comment: The comment should be inside the `if` block.